kiwi/obs.py

```diff
@@ -0,0 +1,240 @@
+from kiwi.runtime_config import RuntimeConfig
+from kiwi.solver.repository.base import SolverRepositoryBase
+from kiwi.system.uri import Uri
+from kiwi.command import Command
+
+from kiwi.exceptions import (
+    KiwiUriOpenError,
+    KiwiOBSBuildInfoError,
+    KiwiOBSProjectError,
+    KiwiOBSSourceError
+)
+
+log = logging.getLogger('kiwi')
+
+
+class OBS:
+    """
+    **Implements methods to access the Open Build Service API**
+    """
+    def __init__(
+        self, image_path: str, user: str, password: str,
+        ssl_verify: bool, profiles: List[str], arch: str, repo: str
+    ):
+        """
+        Initialize OBS API access for a given project and package
+
+        :param str image_path: OBS project/package path
+        :param str user: OBS account user name
+        :param str password: OBS account password
+        :param list profiles: OBS image multibuild profile list
+        :param str arch: OBS architecture
+        :param str repo: OBS image package build repository name
+        :param str api_server:
+            OBS api server to use, defaults to https://api.opensuse.org/build
+        """
+        runtime_config = RuntimeConfig()
+        try:
+            (self.project, self.package) = image_path.split(os.sep)
+        except ValueError:
+            raise KiwiOBSProjectError(f'Invalid image path: {image_path}')
+        self.user = user
+        self.profile = profiles[0] if profiles else None
+        self.password = password
+        self.ssl_verify = ssl_verify or True
+        self.arch = arch or 'x86_64'
+        self.repo = repo or 'images'
+        self.api_server = runtime_config.get_obs_api_server_url()
+
+    def fetch_obs_image(self, checkout_dir) -> str:
+        """
+        Fetch image description from the obs project
+
+        :return: temp local path to image description
+
+        :return: path to local image checkout
+
+        :rtype: str
+        """
+        log.info('Checking out OBS project:')
+        if os.path.exists(checkout_dir):
+            raise KiwiOBSSourceError(
+                f'OBS source checkout dir: {checkout_dir!r} already exists'
+            )
+        log.info(f'--> {self.project}/{self.package}')
+        package_link = os.sep.join(
+            [
+                self.api_server, 'source',
+                self.project, self.package
+            ]
+        )
+        request = self._create_request(package_link)
+        package_source_xml_tree = OBS._import_xml_request(request)
+        package_source_contents = package_source_xml_tree.getroot().xpath(
+            '/directory/entry'
+        )
+        if not package_source_contents:
+            raise KiwiOBSSourceError(
+                f'OBS source for {self.package!r} package is empty'
+            )
+        source_files = []
+        for entry in package_source_contents:
+            source_files.append(entry.get('name'))
+
+        if '_service' in source_files:
+            raise KiwiOBSSourceError(
```

Comment:
This will raise in any sort of `_service` file exists in sources right? I would not do so. I would just throw a warning, it is common practice in OBS having the `_service` file in sources with all the services in `disabled` mode. So in case there is the presence of a `_service` I would just throw a big warning to let the user now the _service file won't be processed and if this is needed for the build it is most likely to fail at some point.

Reply:
yes makes sense. I would even go that far and resolve the _service if it uses ```<service name="obs_scm">``` and ```<param name="scm">git</param>``` That should be really use to implement. In any other case we just take the file as you suggested